substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -1860,6 +1941,41 @@ impl<T: Config> StakingInterface for Pallet<T> {
 	}
 }
 
+impl<T: Config> sp_staking::StakingUnsafe for Pallet<T> {
+	fn migrate_to_virtual_staker(who: &Self::AccountId) {
```

Comment:
Thinking out loud here, but I guess we don't want to allow the callers to do the reverse operation through this trait? e.g. `migrate_to_non_virtual_staker`. This can easily be done through unbonding the virtual account and bonding with staking though, so probably better to not implement it directly.

Reply:
Yeah we shouldn't. For practical purposes, `DelegateStake` is the v2 strategy and we should never really be going back. But technically yeah its possible to write a migration to iterate over delegators and move funds to this virtual ledger account.

Reply:
> we don't want to allow the callers to do the reverse operation
>
> This can easily be done through unbonding the virtual account and bonding with staking

We should surely not *disallow* this, and we probably don't even have a way to disallow it even if we wanted to. But yes, I agree that we shouldn't introduce unneeded functionality in the interface.

> `DelegateStake` is the v2 strategy and we should never really be going back

Our best bet is to incentivize users to use v2 through superior functionality/performance. If there was nothing to gain from using v2, we probably wouldn't be building it, but I think it's clear there's value here and people won't go back.

Reply:
> Yeah we shouldn't. For practical purposes, `DelegateStake` is the v2 strategy and we should never really be going back. But technically yeah its possible to write a migration to iterate over delegators and move funds to this virtual ledger account.

Actually had to do the reverse op for testing, i.e. setting up a pool that has not been migrated yet.

Reply:
So semantically, a normal staker and virtual staker have all the same data, except: 

- real stakers have a lock, but no entry in `VirtualStakers`
- virtual stakers have no lock, but an entry in `VirtualStakers`.

Reply:
As for the rest, I would just encourage everyone to remember that these are more or less internal APIs so I would generally opt for simplicity rather than possibly over-engineering them.

Moreover, once pools' participation in governance + stake tracker is done, I would say the staking system is "safe and feature complete", and at that point we can freeze the existing code, and decide if we want to iterate further on it, or if need be write a new version from scratch. Given this possibility, I think it is acceptable to take a bit of compromise and add even a tiny bit more of debt to this code for the sake of simplicity.